zstd-safe/fuzz/fuzz_targets/zstd_fuzzer.rs

```
@@ -0,0 +1,87 @@
+        if let Ok(written) = zstd_safe::compress(&mut buffer[..], data, level) {
+            let compressed = &buffer[..written];
+            let mut decompressed = vec![0u8; 2048];
+            let _ = zstd_safe::decompress(&mut decompressed[..], compressed).unwrap_or_else(|_| 0);
+        }
+    }
+
+    // Fuzz compression and decompression with CCtx
+    let mut cctx = zstd_safe::CCtx::default();
+    if let Ok(written) = cctx.compress(&mut buffer[..], data, 3) {
+        let compressed = &buffer[..written];
+        let mut dctx = zstd_safe::DCtx::default();
+        let mut decompressed = vec![0u8; 2048];
+        let _ = dctx.decompress(&mut decompressed[..], compressed).unwrap_or_else(|_| 0);
+    }
+
+    // Fuzz compression and decompression on dict
+    let dict = b"sample dictionary for zstd fuzzing";
+    let mut cctx_dict = zstd_safe::CCtx::default();
+    if let Ok(written) = cctx_dict.compress_using_dict(&mut buffer[..], data, dict, 3) {
+        let compressed = &buffer[..written];
+
+        let mut dctx_dict = zstd_safe::DCtx::default();
+        let mut decompressed = vec![0u8; 2048];
+        let _ = dctx_dict.decompress_using_dict(&mut decompressed[..], compressed, dict).unwrap_or_else(|_| 0);
+    }
+
+    // Fuzz compression and decompression with streaming
+    let mut cctx_stream = zstd_safe::CCtx::default();
+    let mut dctx_stream = zstd_safe::DCtx::default();
+    let mut in_buffer = zstd_safe::InBuffer::around(data);
+    let mut out_buffer = zstd_safe::OutBuffer::around(&mut buffer[..]);
+
+    if let Ok(_) = cctx_stream.compress_stream(&mut out_buffer, &mut in_buffer) {
+        let mut decompressed_stream = vec![0u8; 2048];
+        let mut out_buffer_stream = zstd_safe::OutBuffer::around(&mut decompressed_stream[..]);
+        let mut in_buffer_stream = zstd_safe::InBuffer::around(out_buffer.as_slice());
+        let _ = dctx_stream.decompress_stream(&mut out_buffer_stream, &mut in_buffer_stream).unwrap_or_else(|_| 0);
```

This will silently ignore any decompression failure, is that intended?

---

I am thinking if we provide random bytes for fuzzing, there is a high possibility to result in decompression failure and those failure is kinda expected. Thus it is not the target of fuzzing and so I decided to ignore them.